examples/05_tool_inference_lock/ex_tool_inference_lock/inference_lock_helper.rb

```diff
@@ -0,0 +1,122 @@
+module Examples
+  module ToolInferenceLock
+    # Helper object to lock tool inference using Shift or arrow keys.
+    class InferenceLockHelper
+      # Create InferenceLockHelper object.
+      #
+      # @para tool [Object] The SketchUp Ruby tool using this helper object.
+      # @param active_ip [Sketchup::InputPoint]
+      # @param reference_ip [Sketchup::InputPoint]
+      def initialize(tool, active_ip, reference_ip = active_ip)
```

Comment:
I think this class should own the InputPoints used. Right now the tool and this class shares ownership - which breaks single responsibility principle.

The tool can access the input point via accessors of this class. This class is after all what manages the state of the input point and inferencing.

Reply:
Inferencing lock isn't a state of the InputPoint, but of the view. See my previous comment.